lib/node_modules/@stdlib/_tools/changelog/generate/lib/breaking_changes.js

```diff
@@ -54,13 +86,39 @@ var STDLIB_GITHUB_URL = 'https://github.com/stdlib-js/stdlib/commit';
 * // returns '-   [`abcdef1`](https://github.com/stdlib-js/stdlib/commit/abcdef1234567890): beep'
 */
 function formatBreakingChange( note ) {
-	var parts = note.text.split( '\n' );
-	var hash = trim( note.hash );
-	var out = '-   [`'+hash.substring( 0, 7 )+'`]('+STDLIB_GITHUB_URL+'/'+hash+'): '+parts[ 0 ];
-	if ( parts.length > 1 ) {
-		out +='\n\n';
-		out += '    -   ';
-		out += parts.slice( 1 ).join( '\n        ' );
+	var contentLines;
+	var trimmedLine;
+	var firstLine;
+	var restLines;
+	var textLines;
+	var hash;
+	var out;
+	var i;
+
+	textLines = note.text.split( reEOL.REGEXP );
+	contentLines = stripYamlBlocks( textLines );
+	firstLine = '';
+	restLines = [];
+	hash = trim( note.hash );
+
+	for ( i = 0; i < contentLines.length; i++ ) {
+		trimmedLine = trim( contentLines[i] );
+		if ( trimmedLine !== '' ) {
```

Comment:
What if I have a commit body with empty lines between content?

```text
feat: foo bar

This commit does X, Y, Z.

BREAKING CHANGE: do beep

To migrate, users should do A, B, C.

---
...YAML
---
```

Reply:
@kgryte This function does not operate on commit bodies, but only the parsed "Breaking Change" note.

Reply:
Okay, but still. After a `BREAKING CHANGE` line, why should we compress the content which follows?

Reply:
I've updated the example above accordingly. Previously, the logic would have preserved empty lines. Now those are removed.

Reply:
Hmm, yeah... The trimming here was to ensure that any leading "empty" lines containing only whitespace characters will be removed. 

To implement "via a slice by finding the first non-empty string index and the last non-empty string index", is it safe to assume that we don't have such lines?

Reply:
> is it safe to assume that we don't have such lines?

I am not sure which lines you are asking about. Leading empty lines? Trailing empty lines? Empty lines between paragraphs of content?

Reply:
After further conversation, looks like we can update this logic to simply squeeze multiple empty lines into a single empty line, so that we are not making assumptions regarding the location of YAML blocks in commit message bodies.

Reply:
Done.